### PR TITLE
DO NOT MERGE: REP-3502 Make sure JAXB lists with a Default defined actually work as intended

### PR DESCRIPTION
**Pull request by wdschei**

### CODE REVIEW

**Comment by dmnjohns** [comment]
Needs a rebase.

**Comment by dmnjohns** [comment]
Oh, and as for QE review, we should probably add a functional test proving that we can let the HTTP methods attribute default and still get rate limited. Prove the change you made to production code, at least.

File: repose-aggregator/components/filters/translation-filter/src/test/java/org/openrepose/filters/translation/TranslationFilterHandlersTest.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        trans2.setTranslatedContentType(APPLICATION_XML_VALUE);
+        trans2.setStyleSheets(sheets);
+
+        responseTranslations.getResponseTranslation().add(trans2);
+
+        config.setRequestTranslations(requestTranslations);
+        config.setResponseTranslations(responseTranslations);
+        filter.configurationUpdated(config);
+//        assertEquals(1, filter.getRequestProcessors().size());
+//        assertEquals(1, filter.getResponseProcessors().size());
```
Comment (wdschei) [inline repose-aggregator/components/filters/translation-filter/src/test/java/org/openrepose/filters/translation/TranslationFilterHandlersTest.java]: I don't exactly know. I just ripped it out of the original into its own test file so it wasn't crazy big.

File: repose-aggregator/components/filters/translation-filter/src/test/java/org/openrepose/filters/translation/TranslationFilterMethodsTest.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    private HttpServletRequestWrapper httpServletRequestWrapper;
+    private HttpServletResponseWrapper httpServletResponseWrapper;
+
+    @BeforeClass
+    public static void setupSpec() throws Exception {
+        contentRemoveMe = IOUtils.toByteArray(TranslationFilterMethodsTest.class.getResourceAsStream("/remove-me-element.xml"));
+    }
+
+    @Before
+    public void setup() throws Exception {
```
Comment (wdschei) [inline repose-aggregator/components/filters/translation-filter/src/test/java/org/openrepose/filters/translation/TranslationFilterMethodsTest.java]: Yep. I agree this is not the best, but I was making them all behave similarly for continuity.

File: repose-aggregator/components/filters/header-normalization-filter/src/test/scala/org/openrepose/filters/headernormalization/HeaderNormalizationFilterTest.scala [hunk 3]
```diff
@@ -455,6 +455,22 @@ class HeaderNormalizationFilterTest extends FunSpec with BeforeAndAfterEach with
...
         verify(meter, never()).mark(any())
       }
+
+      it(s"will NOT update metrics when a request method does not match any config target on the ${requestResponseTypeWithStyleToString(target, newStyle)}") {
+        val preHeaders = Seq(("legit-header", "such-value"))
+        target match {
+          case RequestTarget => addRequestHeaders(preHeaders)
+          case ResponseTarget => addResponseHeadersOnDoFilter(preHeaders)
+        }
+        servletRequest.setRequestURI("/v2/fries")
```
Comment (wdschei) [inline repose-aggregator/components/filters/header-normalization-filter/src/test/scala/org/openrepose/filters/headernormalization/HeaderNormalizationFilterTest.scala]: Yep. It sure should. Updated.

File: repose-aggregator/components/filters/valkyrie-authorization-filter/src/test/scala/org/openrepose/filters/valkyrieauthorization/ValkyrieAuthorizationFilterTest.scala [hunk 4]
```diff
@@ -249,7 +249,7 @@ class ValkyrieAuthorizationFilterTest extends FunSpec with BeforeAndAfterEach wi
         setMockAkkaBehavior("someTenant", request.headers.getOrElse("X-Contact-Id", "ThisIsMissingAContact"), valkyrie.code, valkyrie.payload)
 
         val filter: ValkyrieAuthorizationFilter = new ValkyrieAuthorizationFilter(mock[ConfigurationService], akkaServiceClientFactory, mockDatastoreService)
-        filter.configurationUpdated(createGenericValkyrieConfiguration(null))
+        filter.configurationUpdated(createGenericValkyrieConfiguration(null, httpMethods = List[HttpMethod]()))
```
Comment (wdschei) [inline repose-aggregator/components/filters/valkyrie-authorization-filter/src/test/scala/org/openrepose/filters/valkyrieauthorization/ValkyrieAuthorizationFilterTest.scala]: Yep. That is cleaner. Updated.

File: repose-aggregator/components/services/rate-limiting-service/src/test/groovy/org/openrepose/core/services/ratelimit/RateLimitingServiceImplTest.groovy [hunk 5]
```diff
@@ -89,8 +94,12 @@ public class RateLimitingServiceImplTest extends RateLimitServiceTestContext {
 
         queryParamLimitGroup.getLimit().add(newLimitConfig("query-param-test", "*", ".*", methods, ["index"].asList()))
 
+        defaultMethodsLimitGroup.getLimit().add newLimitConfig("methods-test-empty", "/methods/empty/*", "/methods/empty/.*", Collections.emptyList(), queryNames)
+        defaultMethodsLimitGroup.getLimit().add newLimitConfig("methods-test-all", "/methods/all/*", "/methods/all/.*", Collections.singletonList(HttpMethod.ALL), queryNames)
```
Comment (wdschei) [inline repose-aggregator/components/services/rate-limiting-service/src/test/groovy/org/openrepose/core/services/ratelimit/RateLimitingServiceImplTest.groovy]: Yep. I don't know why I did that or if some weird formatting thing did it. Updated.

File: repose-aggregator/components/filters/translation-filter/src/test/java/org/openrepose/filters/translation/TranslationFilterMethodsTest.java [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+        RequestTranslation trans1 = new RequestTranslation();
+        trans1.setAccept(APPLICATION_XML_VALUE);
+        trans1.setContentType(APPLICATION_XML_VALUE);
+        trans1.setTranslatedContentType(APPLICATION_XML_VALUE);
+        trans1.setStyleSheets(sheets);
+        trans1.getHttpMethods().add(HttpMethod.POST);
+        requestTranslations.getRequestTranslation().add(trans1);
+
+        ResponseTranslations responseTranslations = new ResponseTranslations();
+        responseTranslations.getResponseTranslation().add(new ResponseTranslation());
```
Comment (wdschei) [inline repose-aggregator/components/filters/translation-filter/src/test/java/org/openrepose/filters/translation/TranslationFilterMethodsTest.java]: This was a copy/pasta from the other tests and wasn't completely removed. Updated.

File: repose-aggregator/components/filters/translation-filter/src/test/java/org/openrepose/filters/translation/TranslationFilterMethodsTest.java [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+        mockRequest.setContent(contentRemoveMe);
+        httpServletRequestWrapper = new HttpServletRequestWrapper(mockRequest);
+        httpServletResponseWrapper = new HttpServletResponseWrapper(mockResponse, ResponseMode.MUTABLE, ResponseMode.MUTABLE);
+
+        HandleRequestResult handleRequestResult = filter.handleRequest(httpServletRequestWrapper, httpServletResponseWrapper);
+        String actual = IOUtils.toString(handleRequestResult.getRequest().getInputStream());
+        String expected = new String(contentRemoveMe);
+
+        assertThat(actual, isSimilarTo(expected));
+    }
```
Comment (wdschei) [inline repose-aggregator/components/filters/translation-filter/src/test/java/org/openrepose/filters/translation/TranslationFilterMethodsTest.java]: We should and it does now.
